alloc-customization-point.html

```diff
@@ -165,11 +201,100 @@ <h2>About the customization points, request for feedback/decision</h2>
   type. In the second approach, it needs only uses_allocator. Types
   like pmr::vector don't care, but types that change their internal
   storage and API based on this decision do care.
+</p>  
+  
+  
+</p>  On further consideration, we concluded that, if a type can convert from a selected allocator
+  to a domain specific allocator, and does so by either specialising the __domain_allocator_traits trait 
+  or providing the domain_alloc_rebind function, then the intentions is to have this type behave as if it 
+  "uses_allocator" and therefore the definition of <tt>uses_allocator</tt> should take such a conversion into account. 
+  This allows such a type to get all the benefits of being uses_alloctor, not just the uses_alloctor construction.
+
 </p>
+
 <p>
   LEWG should decide whether uses_allocator should depend on
   __domain_allocator_traits. 
 </p>
 
+<h2>Proposed wording</h2>
+
+
+<p>
+Insert new section after [allocator.uses.trait]:
+</p>
+
+ <h3>Class template domain_allocator_traits</h3>
+<code>namespace std {
+         template &lt;class T, class Alloc&gt; struct domain_allocator_traits {};
+         }
+         </code>
+</p></blockquote>
+<p><tt>domain_allocator_traits</tt> provides a means to get uses-allocator construction with allocator types other than the
+type specific allocator. Meets the <i>Cpp17BinaryTypeTrait</i> requirements ([meta.rqmts]). </p>
+
+<p>If <tt>domain_allocator_traits&lt;class T, class Alloc&gt;::value</tt> is <tt>true</tt>,  there will exist a 
+function object member of <tt>domain_allocator_traits</tt> called <tt>rebind</tt> which when invoked with a parameter of type Alloc
+will return type DomainAlloc, and type T will have trailing_allocator convention constructors using DomainAlloc. 
+Otherwise, <tt>domain_allocator_traits&lt;class T, class Alloc&gt;::value</tt> will be <tt>false</tt>. </p>
+
+<p>Implementation shall provide a definition of <tt>domain_allocator_traits&lt;class T, class Alloc&gt</tt> that is 
```

Comment:
I'm confused. That's not how i implemented it. We don't check for domain_allocatr_type, we check if there is a rebind function and if it is callable.

Reply:
Ah. We need to rename rebind. It's too short and too general. I envisioned that we would check for a nested opt-in typedef like uses-allocator already does, and then additionally call a member function, but we can collapse those things into one; detect the member function and call it if it's there. Name it domain_allocator_convert() for now, mark it as "TBD bikeshed by LEWG" ?

Reply:
Also, there are existing/previously-existing rebind members that are types, not functions. We shouldn't confuse this function with them.

Reply:
We have already collapsed those things into one, but the detestion does also insure it's callable - just in case. The rebind member function we look for is  currently called doman_allocator_rebind. I didn't call it just rebind, that would be silly :) Do you still want to rename it domain_allocator_convert ? Will mark it for bikeshedding, though

Reply:
The detection should not ensure callability; that's wrong. If you have a nested domain_allocator_convert (let's rename it to that, yes), then it's going to be attempted whether it's callable or not; we don't SFINAE away from it.

Reply:
renamed. Removed the cool callability detection (sigh)

Reply:
added a nested type called domain_allocator_conversion_source for the purposes of determining the type allocator from which we can get domain allocator.